kuiper-game/Domain/CelestialBodies/CelestialBody.cs

```diff
@@ -0,0 +1,130 @@
+using System;
+using System.Collections.Generic;
+using System.Numerics;
+
+namespace Kuiper.Domain.CelestialBodies
+{
+    public abstract class CelestialBody
+    {
+        private const int AUINKM =  149597871; // 1 AU in KM. Maybe this belongs somewhere else?
+        public string Name
+        {
+            get
+            {
+                return name;
+            }
+        }
+
+        public double OrbitRadius
+        {
+            get
+            {
+                return orbitRadius;
+            }
+        }
+
+        public double Velocity
+        {
+            get
+            {
+                return velocity;
+            }
+        }
+
+        public double OriginDegrees
+        {
+            get
+            {
+                return originDegrees;
+            }
+        }
+
+        public Vector2 GetPosition(TimeSpan gametimePassed)
+        {
+            if (parent == null) return new Vector2(0,0);
+            var velocityInRadiansPerSecond = 
+                velocity / (OrbitRadius * AUINKM);
+            var positionInRadians = originDegrees * (Math.PI/180);
+            positionInRadians += (velocityInRadiansPerSecond * gametimePassed.TotalSeconds);
+
+            var x = parent.GetPosition(gametimePassed).X + Math.Cos(positionInRadians) * OrbitRadius;
+            var y = parent.GetPosition(gametimePassed).Y + Math.Sin(positionInRadians) * OrbitRadius;
+
+            return new Vector2((float)x, (float)y);
+        }
+
+        public CelestialBody Parent
+        {
+            get
+            {
+                return parent;
+            }
+        }
+
+        public IEnumerable<CelestialBody> Satellites
+        {
+            get
+            {
+                return satellites;
+            }
+        }
+
+        private string name;
+        private double orbitRadius;
+        private double velocity;
+        private double originDegrees;
+        private CelestialBody parent;
+        private List<CelestialBody> satellites;
+
+        public static CelestialBody Create(string name, double orbitRadius, double velocity, double originDegrees, CelestialBody parent, CelestialBodyType bodyType)
```

Comment:
For the mining gamesystem, and perhaps later on for static installations, is it possible to simplify this method?

Gravity isn't a thing so you can't create an invalid orbit, like too high a velocity for a stabile orbit at a given orbitRadius. 

I suppose velocity can be inferred from the chosen orbitRadius? As in a sattelite at radius Y should have a velocity of X to be realistic. The closer you are to the sun, the higher the velocity (as in how quickly your degrees change)

Reply:
Could perhaps be an overload with no velocity that then decides based on radius.
I definitely see orbiting stations as just another satellite of some type that inherits from CelestialBody.

Reply:
Ok @mifriis, I've added two overloads to `Create()`.
You can now create a body without a velocity and also one without velocity AND originDegrees.

Velocity will be calculated to look similar to our solarsystem.
Origin degrees will just be random.